build.gradle

```diff
@@ -236,7 +236,6 @@ project(':samza-aws') {
     compile project(":samza-core_$scalaVersion")
     compile "org.slf4j:slf4j-api:$slf4jVersion"
     runtime "org.apache.httpcomponents:httpclient:4.5.2"
```

Comment:
Modules `samza-yarn`, `samza-autoscaling` are using httpClient version `4.4.1` through `httpClientVersion` variable defined in dependency-versions.gradle.

Since httpclient `4.5.x` version is backwards compatible with `4.4.x`, I think it will be better to use one consistent version across all samza modules(Either update other modules to use 4.5.x or downgrade this one to 4.4.x).

Please change if this makes sense.

Reply:
My observation was that the version backwards incompatibility problem is on httpcore (between 4.4.4 and 4.4.5), not httpclient. The httpclient:4.5.2 and httpclient:4.4.1 both depend on versions of httpcore which are 4.4.4 or lower, so there doesn't seem to be a backwards incompatibility problem there.
I do agree that it would be nice to have consistent versions, but I feel that is out of scope here and can be done in a separate PR which pulls in new versions for a bunch of other dependencies as well.